.github/workflows/linters.yml

```diff
@@ -8,25 +8,41 @@ name: Format code using black and isort
 
 on:
   push:
-    branches: [ "develop", "issue23-precommithooks" ]
-
+    branches-ignore:
+          - master
 jobs:
   linters:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Format code with black 
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install dependencies
         run: |
-          pip install black==21.12b0
-          black .
-      - name: Format code with isort
+          python -m pip install --upgrade pip
+          python -m pip install tomli
+          python -m pip install black
+          python -m pip install isort
+          python -m pip install click
+      - name: Extract version requirements from pyproject.toml and install
         run: |
-          pip install  isort==5.10.1
-          isort .
+          export click_version=$(python -c 'import tomli; print([line for line in tomli.load(open("pyproject.toml","rb"))["project"]["optional-dependencies"]["dev"] if "click" in line][0])')
+          export black_version=$(python -c 'import tomli; print([line for line in tomli.load(open("pyproject.toml","rb"))["project"]["optional-dependencies"]["dev"] if "black" in line][0])')
+          export isort_version=$(python -c 'import tomli; print([line for line in tomli.load(open("pyproject.toml","rb"))["project"]["optional-dependencies"]["dev"] if "isort" in line][0])')
+          echo "Click version: $click_version"
+          echo "Black version: $black_version"
+          echo "Isort version: $isort_version"
+          python -m pip install $click_version $black_version $isort_version
+      - name: Format code using black
+        run: black .
+      - name: Sort import statements using isort
+        run: isort src tests doc noxfile.py
```

Comment:
update pyproject.toml to include the paths for isort to search

Reply:
done